addons/survey/models/survey_survey.py

```diff
@@ -626,14 +626,22 @@ def prepare_result(self, question, current_filters=None):
     @api.model
     def get_input_summary(self, question, current_filters=None):
         """ Returns overall summary of question e.g. answered, skipped, total_inputs on basis of filter """
-        current_filters = current_filters if current_filters else []
-        result = {}
-        search_line_ids = current_filters if current_filters else question.user_input_line_ids.ids
-
-        result['answered'] = len([line for line in question.user_input_line_ids if line.user_input_id.state != 'new' and not line.user_input_id.test_entry and not line.skipped])
-        result['skipped'] = len([line for line in question.user_input_line_ids if line.user_input_id.state != 'new' and not line.user_input_id.test_entry and line.skipped])
-
-        return result
+        current_filters = [
+            line
+            for line in self.env["survey.user_input_line"].browse(current_filters)
+            if line.user_input_id.state != 'new' and not line.user_input_id.test_entry
+        ] if current_filters else []
+
+        line_ids = current_filters or self.env["survey.user_input_line"].search([
+            '&', '&',
+            ('question_id', '=', question.id),
+            ('user_input_id.test_entry', '=', False),
+            ('user_input_id.state', '!=', 'new')
+        ])
```

Comment:
Looks like we could simplify a bit?

```py
domain = [
    ('user_input_id.test_entry', '=', False),
    ('user_input_id.state', '!=', 'new'),
    ('question_id', '=', question.id)
]

if current_filters:
    domain = expression.AND([[('id', 'in', current_filters)], domain])

line_ids = self.env["survey.user_input_line"].search(domain)
...
```

Ideally this would be doable with 2 read_group requests grouped on user_input_id with the appropriate domain, but probably not necessary for a bug fix, will make the diff bigger.

Reply:
Yes, nice :+1